CMakeLists.txt

```diff
@@ -146,7 +147,7 @@ include_directories(Framework/API/inc)
 include_directories(Framework/Types/inc)
 include_directories(Framework/NexusGeometry/inc)
 
-set(CORE_MANTIDLIBS Kernel HistogramData Indexing Beamline Geometry API Types)
+set(CORE_MANTIDLIBS Kernel Json HistogramData Indexing Beamline Geometry API Types)
```

Comment:
Do we need this `CORE_MANTIDLIBS`? Can we put it in the Kernel public link libraries so that CMake will forward it automatically?